InnerEye-DataQuality/InnerEyeDataQuality/datasets/cifar10_utils.py

```diff
@@ -0,0 +1,165 @@
+    """
+    :param labels: Complete label array that is split into two subsets, reference and noisy test set.
+    :param reference_split: Ratio of total samples that will be put in the reference set.
+    :param shuffle: If set to true, rows of the label matrix are shuffled prior to split.
+    :param seed: Random seed used in sample shuffle
+    """
+    # Create two testing sets from CIFAR10H Test Samples (10k - 10 classes)
+    num_samples = labels.shape[0]
+    num_samples_set1 = int(num_samples * reference_split)
+    perm = np.random.RandomState(seed=seed).permutation(num_samples) if shuffle else np.array(range(num_samples))
+    d_set1 = labels[perm[:num_samples_set1], :]
+    d_set2 = labels[perm[num_samples_set1:], :]
+
+    return d_set1, d_set2, perm
+
+
+def get_cifar10h_labels(reference_split: float = 0.5,
+                        shuffle: bool = True,
+                        immutable: bool = True) -> Tuple[np.ndarray, np.ndarray]:
+    """
+    :param reference_split: The sample ratio between gold standard test set and full test set
+    :param shuffle: Shuffle samples prior to split.
+    :param immutable: If set to True, the returned arrays are only read only.
+    """
+    cifar10h_counts = download_cifar10h_data()  # Num_samples x n_classes
+    d_split1_standard, d_split2_complete, permutation = split_dataset(cifar10h_counts, reference_split, shuffle)
+    d_split1_permutation = permutation[:d_split1_standard.shape[0]]
+
+    if immutable:
+        d_split1_standard.setflags(write=False)
+        d_split2_complete.setflags(write=False)
+
+    return d_split1_standard, d_split1_permutation
+
+
+def download_cifar10h_data() -> np.ndarray:
+    """
+    Pulls cifar10h label data stream and returns it in numpy array.
+    """
+
+    url = 'https://raw.githubusercontent.com/jcpeterson/cifar-10h/master/data/cifar10h-counts.npy'
+    response = requests.get(url)
+    response.raise_for_status()
+    if response.status_code == requests.codes.ok:
+        cifar10h_data = np.load(io.BytesIO(response.content))
+    else:
+        raise ValueError('CIFAR10H content was not found.')
+
+    return cifar10h_data
+
+
+def download_cifar10_data() -> Path:
+    """
+    Download CIFAR10 dataset and returns path to the test set
+    """
+    import wget
+    local_path = Path.cwd() / 'InnerEyeDataQuality' / 'downloaded_data'
+    local_path_to_test_batch = local_path / 'cifar-10-batches-py/test_batch'
+
+    if not local_path_to_test_batch.exists():
+        local_path.mkdir(parents=True, exist_ok=True)
+        url = 'https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz'
+        path_to_tar = local_path / 'cifar10.tar.gz'
+        wget.download(url, str(path_to_tar))
+        tf = tarfile.open(str(path_to_tar))
+        tf.extractall(local_path)
+        os.remove(path_to_tar)
+
+    return local_path_to_test_batch
+
+
+def get_cifar10_label_names(file: Optional[Path] = None) -> List[str]:
+    """
+    TBD
```

Comment:
can we either remove that completely, or fill in? Same for other occurrences of TBD